lib/Solver/SolverCmdLine.cpp

```diff
@@ -112,17 +112,19 @@ cl::opt<bool> UseAssignmentValidatingSolver(
     cl::desc("Debug the correctness of generated assignments (default=false)"),
     cl::cat(SolvingCat));
 
-
-void KCommandLine::HideOptions(llvm::cl::OptionCategory &Category) {
+void KCommandLine::KeepOnlyCategories(
+    std::set<llvm::cl::OptionCategory *> categories) {
   StringMap<cl::Option *> &map = cl::getRegisteredOptions();
 
   for (auto &elem : map) {
     for (auto &cat : elem.second->Categories) {
-      if (cat == &Category) {
+      if (categories.find(cat) == categories.end()) {
         elem.second->setHiddenFlag(cl::Hidden);
       }
     }
   }
+  map["help"]->setHiddenFlag(cl::NotHidden);
+  map["version"]->setHiddenFlag(cl::NotHidden);
```

Comment:
We can skip setting and unsetting the hidden flag for these options by checking `elem.first() == "help" || elem.first() == "version"` in the loop above